File: Array/260.Single_Number_III/C++/260.Single_Number_III.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int> & nums) 
    {
        //====================================//
        // Input type:                        //
        //   - vector<int> ref. (nums vector) //
        // Return type:                       //
        //   - vector<int> (result vector)    //
        //====================================//

        /*Initialize*/
        ///// Length of nums vector /////
        int len_nums = nums.size();

        ///// Record dictionary, hashmap /////
        unordered_map<int, int> record_map;

        ///// Result vector /////
        vector<int> res_vect;


        /*Hashmap-based loop traversal*/
        ///// Step 1: Record indexed-value with dictionary, hashmap /////
        for (int nums_idx = 0; (nums_idx < len_nums); (nums_idx++)) //Whole
        {
            ((record_map[nums[nums_idx]])++); //Keep updating, recording

        } //Whole

        ///// Step 2: Looped-traversal for the recorded dictionary, hashmap /////
        for (auto & record_ele: record_map) //Whole
        {
            ///// Check if the current summary counter matched conditions or not /////
            if (record_ele.second == 1) { res_vect.emplace_back(record_ele.first); } //Keep updating, recording
            else { ; }

            ///// Check if the current length matched conditions or not /////
            if (res_vect.size() == 2) { break; }

        } //Whole

        return res_vect;
    }
};

File: Array/260.Single_Number_III/Python/260.Single_Number_III.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        #================================#
        # Hashmap-based traversal method #
        #================================#

        ############
        #Initialize
        ##### Length of nums array #####
        len_nums = len(nums)

        ##### Record dictionary/hashmap #####
        record_dict = {}

        ##### Result array #####
        res_arry = []


        ##############################
        #Hashmap-based loop traversal
        ##### Step 1: Record indexed-value with dictionary/hashmap #####
        for nums_idx in range(len_nums):

            ##### Check if the current indexed-value existed or not #####
            if (nums[nums_idx] not in record_dict):
                record_dict[nums[nums_idx]] = 1 #Keep updating/recording
            else:
                record_dict[nums[nums_idx]] += 1 #Keep updating/recording

        ##### Step 2: Looped-traversal for the recorded dictionary/hashmap #####
        for record_key in record_dict:

            ##### Check if the current value matched conditions or not #####
            if (record_dict[record_key] == 1):
                res_arry.append(record_key) #Keep updating/recording
            else:
                pass

            ##### Check if the current length matched conditions or not #####
            if (len(res_arry) == 2):
                break

        return res_arry
